developer/src/kmc-package/src/compiler/get-file-data.ts

```diff
@@ -0,0 +1,232 @@
+  if(!family) {
+    callbacks.reportMessage(PackageCompilerMessages.Warn_FontNotFoundInFlo({filename: floSource, family: matches.groups.family}));
+    return null;
+  }
+
+  if(!family.distributable) {
+    callbacks.reportMessage(PackageCompilerMessages.Warn_FontFromFloIsNotFreelyDistributable(
+      {filename: floSource, family: matches.groups.family}));
+    // safe to continue
+  }
+
+  // TODO: consider .woff, .woff2 for web font inclusion
+  const ttf = family.defaults?.ttf;
+  if(!ttf) {
+    callbacks.reportMessage(PackageCompilerMessages.Warn_FontInFloDoesNotHaveDefaultTtf({filename: floSource, family: matches.groups.family}));
+    return null;
+  }
+
+  const file = family.files[ttf];
+  /* c8 ignore next 4 */
+  if(!file) {
+    callbacks.reportMessage(PackageCompilerMessages.Warn_FontInFloHasBrokenDefaultTtf({filename: floSource, family: matches.groups.family}));
+    return null;
+  }
+
+  if(!file.url) {
+    callbacks.reportMessage(PackageCompilerMessages.Warn_FontInFloHasNoDownloadAvailable({filename: floSource, family: matches.groups.family}));
+    return null;
+  }
+
+  // we don't use flourl at this time, becase we want a GitHub reference that
+  // can be resolved to a stable URI
+  const ghmatches = GITHUB_URI.exec(file.url) ?? GITHUB_RAW_URI.exec(file.url);
+  /* c8 ignore next 4 */
+  if(!ghmatches) {
+    callbacks.reportMessage(PackageCompilerMessages.Error_FontInFloDoesNotHaveARecognizedGitHubUri({filename: floSource, url: file.url}));
+    return null;
+  }
+
+  return await resolveGitHubToStableRef(callbacks, ghmatches);
+}
+
+async function getFileStableRefFromGitHub(callbacks: CompilerCallbacks, source: string): Promise<string> {
+  const matches = GITHUB_URI.exec(source) ?? GITHUB_RAW_URI.exec(source);
+  if(!matches) {
+    callbacks.reportMessage(PackageCompilerMessages.Error_UriIsNotARecognizedGitHubUri({url: source}));
+    return null;
+  }
+  return await resolveGitHubToStableRef(callbacks, matches);
+}
+
+async function resolveGitHubToStableRef(callbacks: CompilerCallbacks, matches: RegExpExecArray): Promise<string> {
+  let commit: any = null;
+
+  const url = `https://api.github.com/repos/${matches.groups.name}/${matches.groups.repo}/commits/${matches.groups.branch}?path=${matches.groups.path}`;
+  try {
+    commit = await callbacks.net.fetchJSON(url);
+    /* c8 ignore next 8 */
+    if(!commit?.sha) {
+      callbacks.reportMessage(PackageCompilerMessages.Error_CouldNotRetrieveStableUriFromGitHub({url}));
+      return null;
+    }
+  } catch(e) {
+    callbacks.reportMessage(PackageCompilerMessages.Error_CouldNotRetrieveStableUriFromGitHub({url, e}));
+    throw e;
+  }
+
+  return `https://github.com/${matches.groups.name}/${matches.groups.repo}/raw/${commit.sha}/${matches.groups.path}`;
+}
+
+async function getFileDataFromGitHub(callbacks: CompilerCallbacks, inputFilename: string, matches: RegExpExecArray): Promise<KmpCompilerFileDataResult> {
```

Comment:
To keep the order the same as with the previous functions where we define the outer function first and then the function that gets called by the outer function, you could move this function after `getFileDataFromRemote`. Would make it easier to read when reading the whole file.